src/language/schema/__tests__/parser.js

```diff
@@ -0,0 +1,450 @@
+ */
+
+import { expect } from 'chai';
+import { describe, it } from 'mocha';
+import { parseSchema } from '../parser';
+
+function createLocFn(body) {
+  return (start, end) => ({
+    start: start,
+    end: end,
+    source: {
+      body: body,
+      name: 'GraphQL',
+    },
+  });
+}
+
+function printJson(obj) {
+  return JSON.stringify(obj, null, 2);
+}
+
+function typeNode(name, loc) {
+  return {
+    kind: 'NamedType',
+    name: nameNode(name, loc),
+    loc: loc,
+  };
+}
+
+function nameNode(name, loc) {
+  return {
+    kind: 'Name',
+    value: name,
+    loc: loc,
+  };
+}
+
+function fieldNode(name, type, loc) {
+  return fieldNodeWithArgs(name, type, [], loc);
+}
+
+function fieldNodeWithArgs(name, type, args, loc) {
+  return {
+    kind: 'FieldDefinition',
+    name: name,
+    type: type,
+    arguments: args,
+    loc: loc,
+  };
+}
+
+describe('Schema Parser', () => {
+  it('Simple type', () => {
+    var body = `
+type Hello {
+  world: String
+}`;
+    var doc = parseSchema(body);
+    var loc = createLocFn(body);
+    var expected = {
+      kind: 'SchemaDocument',
+      definitions: [
+        {
+          kind: 'TypeDefinition',
+          name: nameNode('Hello', loc(6, 11)),
+          interfaces: [],
+          fields: [
+            fieldNode(
+              nameNode('world', loc(16, 21)),
+              typeNode('String', loc(23, 29)),
+              loc(16, 29)
+            )
+          ],
+          loc: loc(1, 31),
+        }
+      ],
+      loc: loc(1, 31),
+    };
+    expect(printJson(expected)).to.equal(printJson(doc));
+  });
+
+  it('Simple non-null type', () => {
+    var body = `
+type Hello {
+  world: String!
+}`;
+    var loc = createLocFn(body);
+    var doc = parseSchema(body);
+    var expected = {
+      kind: 'SchemaDocument',
+      definitions: [
+        {
+          kind: 'TypeDefinition',
+          name: nameNode('Hello', loc(6, 11)),
+          interfaces: [],
+          fields: [
+            fieldNode(
+              nameNode('world', loc(16, 21)),
+              {
+                kind: 'NonNullType',
+                type: typeNode('String', loc(23, 29)),
+                loc: loc(23, 30),
+              },
+              loc(16, 30)
+            )
+          ],
+          loc: loc(1, 32),
+        }
+      ],
+      loc: loc(1, 32),
+    };
+    expect(printJson(expected)).to.equal(printJson(doc));
+  });
+
+
+  it('Simple type inheriting interface', () => {
+    var body = `type Hello implements World { }`;
+    var loc = createLocFn(body);
+    var doc = parseSchema(body);
+    var expected = {
+      kind: 'SchemaDocument',
+      definitions: [
+        {
+          kind: 'TypeDefinition',
+          name: nameNode('Hello', loc(5, 10)),
+          interfaces: [typeNode('World', loc(22, 27))],
+          fields: [],
+          loc: loc(0, 31),
+        }
+      ],
+      loc: loc(0, 31),
+    };
+    expect(printJson(doc)).to.equal(printJson(expected));
```

Comment:
Let's ultimately define a helper for mocha then in mocha-bootstrap, and perhaps send it upstream to mocha.